The titleID checking is now immensely faster

plato79 opened the pull request:
If the filename has titleID in it, the check now executed before decrypting ticket. Because of that it skips files faster (almost immediately).

Though, the code is a bit dirty now, we need to move these checks to a method I think.

Feel free to remove remarks. Did I say I love putting remarks on my code ;)

Also moved glob.glob to nsz.py to prevent getting output dir contents for every file. Though, I had to use a parameter to move the dir contents to SolidCompress and BlockCompress

nicoboss commented:
The regex "[0-9A-Fa-f]{16}" should be more specific. It should check if the titleid starts with "0100" to avoid potential false positives. After that to the regex the PR is ready to be merged.

In the future we should also extract the version number from the file name and if not found from the file's metadata. This can be done by searching for "v" followed by a multiplier of "65536". The version number is needed so newer updates won't cause false positive overwrite message and an option to remove old updates can be added.

plato79 commented:
Just implemented version checking in nsz.py....

fnmatch was not good enough ( or my patterns suck ) so I moved to re.match for titleID + version number check.

Also changed titleID pattern to '0100[0-9A-Fa-f]{12}'. It should correctly detect titleIDs now.
